tests/playwright-test/only-changed.spec.ts

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { test as baseTest, expect, playwrightCtConfigText } from './playwright-test-fixtures';
+import { execSync } from 'node:child_process';
+
+const test = baseTest.extend({
+  setupRepository: async ({ writeFiles }, use, testInfo) => {
+    const baseDir = testInfo.outputPath();
+
+    const git = (command: string) => execSync(`git ${command}`, { cwd: baseDir });
+
+    await use(async () => {
+      await writeFiles({
+        'a.spec.ts': `
+        import { test, expect } from '@playwright/test';
+        import { answer, question } from './utils';
+        test('fails', () => { expect(question).toBe(answer); });
+      `,
+        'b.spec.ts': `
+        import { test, expect } from '@playwright/test';
+        import { answer, question } from './utils';
+        test('fails', () => { expect(question).toBe(answer); });
+      `,
+        'utils.ts': `
+        export * from './answer';
+        export * from './question';
+      `,
+        'answer.ts': `
+        export const answer = 42;
+      `,
+        'question.ts': `
+        export const question = "???";
+      `,
+      });
+      git(`init --initial-branch=main`);
+      git(`config --local user.name "Robert Botman"`);
+      git(`config --local user.email "botty@mcbotface.com"`);
+      git(`add .`);
+      git(`commit -m init`);
+      return git;
+    });
+  },
+});
+
+test.slow();
+
+test('should detect untracked files', async ({ runInlineTest, setupRepository }) => {
+  await setupRepository();
+
+  const result = await runInlineTest({
+    'c.spec.ts': `
+      import { test, expect } from '@playwright/test';
+      test('fails', () => { expect(1).toBe(2); });
+    `
+  }, { 'only-changed': true });
+
+  expect(result.exitCode).toBe(1);
+  expect(result.failed).toBe(1);
```

Comment:
I'd also check `result.total` to make it obvious that we have only run a single test.

Reply:
I don't see a `result.total`, only `result.passed`. Are you referring to that, or am I missing something?

Reply:
Yeah, `result.passed` is also ok. But we can introduce `result.total` as well. Up to you.

Reply:
added `result.passed` assertions in https://github.com/microsoft/playwright/pull/31727/commits/bc96c8b68c2233f9f435c1b1e93dd6f9a855603e. Had a look at introducing `result.total`, but it wasn't obvious to me wether `result.total` should include skipped tests or not, so I've not added that for now.